packages/lib/src/vis/hooks.ts

```diff
@@ -63,15 +63,11 @@ export function useCameraState<T>(
   const [state, setState] = useState(() => factory(camera, context));
 
   function updateState() {
-    const newState = factory(camera, context);
-
-    if (newState !== state) {
-      setState(newState);
-    }
+    setState(() => factory(camera, context)); // functional update to skip re-renders if state hasn't changed
```

Comment:
Very small simplification: the documentation of [`useState`](https://reactjs.org/docs/hooks-reference.html#usestate) specifies that when doing a functional state update like this (i.e. by passing a function to `setState`), React does not re-render if the state doesn't change. Basically, it does the equality check for us. 

Also, using the current state from `updateState` like this was pretty bad practice. That's exactly what the functional state update pattern is for. Sorry about that.

Reply:
No worries. Each day is a learning experience :sun_with_face: